File: JAVA/442. Find All Duplicates in an Array.java
class Solution {
  public List findDuplicates(int[] nums)
   {
   List res = new ArrayList<>();
   Arrays.sort(nums);
   for(int i=0;i<nums.length-1;i++)
     {
   if(nums[i] == nums[i+1])
     {
   res.add(nums[i]);
    }
   }
  return res;
   }
}

File: C++/442. Find All Duplicates in an Array.cpp
class Solution {
public:
    vector<int> findDuplicates(vector<int>& nums) {
        vector<int> ans; 
        for (auto x : nums) {
            x = abs(x); 
            if (nums[x-1] < 0) ans.push_back(x); 
            nums[x-1] *= -1; 
        }
        return ans; 
    }
};
